ex2/ex2.c

```diff
@@ -5,10 +5,61 @@
 #include <stdio.h>
 #include <unistd.h>
 #include <stdlib.h>
+#include <sys/types.h>
+#include <sys/stat.h>
 
 int main(void)
 {
     // Your code here 
+
+    // open file
+    FILE * fp = fopen("text.txt", "w+");
+    if (!fp) {
+        perror("File opening failed");
+        return EXIT_FAILURE;
+    }
+    mkdir("subdir", 0755);
+
+
+    // read file loop
+    int c;
+    while((c = fgetc(fp)) != EOF) 
+    {
+        putchar(c);
+    }
+
+    // writing to file from parent and child process
+
+    fprintf(fp, "this came from the program\n");
+
+    pid_t pid = fork();
+    if (pid == 0) // Child process
```

Comment:
now, try passing in `fwrite()` statements inside both the child and the parent process to see what turns out in your .txt file